File: app/src/main/java/com/antyzero/smoksmog/ui/screen/ActivityComponent.java
package com.antyzero.smoksmog.ui.screen;

import com.antyzero.smoksmog.google.GoogleModule;
import com.antyzero.smoksmog.ui.screen.about.AboutActivity;
import com.antyzero.smoksmog.ui.screen.history.HistoryActivity;
import com.antyzero.smoksmog.ui.screen.order.OrderActivity;
import com.antyzero.smoksmog.ui.screen.order.dialog.AddStationDialogComponent;
import com.antyzero.smoksmog.ui.screen.start.StartActivity;
import com.antyzero.smoksmog.ui.screen.start.fragment.LocationStationFragmentComponent;
import com.antyzero.smoksmog.ui.widget.StationWidgetConfigureActivity;

import org.jetbrains.annotations.NotNull;

import dagger.Subcomponent;

@Subcomponent(
        modules = {
                ActivityModule.class,
        }
)
public interface ActivityComponent {

    FragmentComponent plus( FragmentModule fragmentModule );

    // Support fragments

    AddStationDialogComponent plus( SupportFragmentModule fragmentModule );

    LocationStationFragmentComponent plus( FragmentModule fragmentModule, GoogleModule googleModule );

    void inject( HistoryActivity activity );

    void inject( AboutActivity activity );

    void inject( StartActivity startActivity );

    void inject( OrderActivity orderActivity );

    void inject(@NotNull PickStation pickStation);

    void inject(@NotNull StationWidgetConfigureActivity stationWidgetConfigureActivity);
}

File: app/src/main/kotlin/com/antyzero/smoksmog/ui/screen/ActivityComponent.kt
package com.antyzero.smoksmog.ui.screen

import com.antyzero.smoksmog.google.GoogleModule
import com.antyzero.smoksmog.ui.screen.about.AboutActivity
import com.antyzero.smoksmog.ui.screen.history.HistoryActivity
import com.antyzero.smoksmog.ui.screen.order.OrderActivity
import com.antyzero.smoksmog.ui.screen.start.StartActivity
import com.antyzero.smoksmog.ui.screen.start.fragment.LocationStationFragmentComponent
import com.antyzero.smoksmog.ui.widget.StationWidgetConfigureActivity
import dagger.Subcomponent

@Subcomponent(modules = arrayOf(ActivityModule::class))
interface ActivityComponent {

    operator fun plus(fragmentModule: FragmentModule): FragmentComponent

    fun plus(fragmentModule: FragmentModule, googleModule: GoogleModule): LocationStationFragmentComponent

    fun inject(activity: HistoryActivity)

    fun inject(activity: AboutActivity)

    fun inject(startActivity: StartActivity)

    fun inject(orderActivity: OrderActivity)

    fun inject(pickStationActivity: PickStationActivity)

    fun inject(stationWidgetConfigureActivity: StationWidgetConfigureActivity)
}
